src/sql/workbench/contrib/notebook/browser/models/diffNotebookInput.ts

```diff
@@ -0,0 +1,91 @@
+import { DiffEditorInput } from 'vs/workbench/common/editor/diffEditorInput';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { FileNotebookInput } from 'sql/workbench/contrib/notebook/browser/models/fileNotebookInput';
+import { INotebookService } from 'sql/workbench/services/notebook/browser/notebookService';
+import { Deferred } from 'sql/base/common/promise';
+import { ILogService } from 'vs/platform/log/common/log';
+
+export class DiffNotebookInput extends SideBySideEditorInput {
+	public static ID: string = 'workbench.editorinputs.DiffNotebookInput';
+	private _notebookService: INotebookService;
+	private _logService: ILogService;
+
+	constructor(
+		title: string,
+		diffInput: DiffEditorInput,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@INotebookService notebookService: INotebookService,
+		@ILogService logService: ILogService
+	) {
+		let originalInput = instantiationService.createInstance(FileNotebookInput, diffInput.primary.getName(), diffInput.primary.resource, diffInput.originalInput as FileEditorInput);
+		let modifiedInput = instantiationService.createInstance(FileNotebookInput, diffInput.secondary.getName(), diffInput.secondary.resource, diffInput.modifiedInput as FileEditorInput);
+		super(title, diffInput.getTitle(), modifiedInput, originalInput);
+		this._notebookService = notebookService;
+		this._logService = logService;
+		this.setupScrollListeners(originalInput, modifiedInput);
+	}
+
+	public getTypeId(): string {
+		return DiffNotebookInput.ID;
+	}
+
+	/**
+	 * Setup scroll listeners so that both the original and modified editors scroll together
+	 * @param originalInput original notebook input
+	 * @param modifiedInput modified notebook input
+	 */
+	private setupScrollListeners(originalInput: FileNotebookInput, modifiedInput: FileNotebookInput): void {
+		Promise.all([originalInput.containerResolved, modifiedInput.containerResolved]).then(() => {
+
+			// Setting container height to 100% ensures that scrollbars will be added when in diff mode
+			originalInput.container.parentElement.style.height = '100%';
+			modifiedInput.container.parentElement.style.height = '100%';
+
+			// Keep track of when original and modified notebooks are shown
+			const originalNotebookEditorShown: Deferred<void> = new Deferred<void>();
+			const modifiedNotebookEditorShown: Deferred<void> = new Deferred<void>();
+
+			// Possible for notebooks to have been shown already, so check this case
+			if (this._notebookService.findNotebookEditor(originalInput.notebookUri)) {
+				originalNotebookEditorShown.resolve();
+			} else if (this._notebookService.findNotebookEditor(modifiedInput.notebookUri)) {
+				modifiedNotebookEditorShown.resolve();
+			}
+
+			// If not already shown, listen for add events
+			this._register(this._notebookService.onNotebookEditorAdd((e) => {
+				if (e.id === originalInput.notebookUri.toString()) {
```

Comment:
This is broken if you're comparing a file to itself FYI. This will never resolve so the scrollbars will never be hooked up. 

Changing it to this worked for me 

```
if (e.notebookParams.input === originalInput) {
	originalNotebookEditorShown.resolve();
} else if (e.notebookParams.input === modifiedInput) {
	modifiedNotebookEditorShown.resolve();
}
```

Reply:
Hmm I don't quite understand the scenario here. Does a file need to be changed to show up in the source control viewlet? How does someone compare the same file to itself?